solr/core/src/java/org/apache/solr/update/processor/UpdateRequestProcessorChain.java

```diff
@@ -339,27 +359,20 @@ public UpdateRequestProcessorFactory get() {
       return lazyFactory;
     }
 
-    public class LazyUpdateRequestProcessorFactory extends UpdateRequestProcessorFactory {
-      private final PluginBag.LazyPluginHolder holder;
-      UpdateRequestProcessorFactory delegate;
+    public static class LazyUpdateRequestProcessorFactory extends UpdateRequestProcessorFactory {
+      private final PluginBag.PluginHolder<UpdateRequestProcessorFactory> holder;
```

Comment:
I think it might be nicer if we declared this as simply Supplier<UpdateRequestProcessorFactory> and then made that PluginHolder thing implement a Supplier<T>?

Reply:
These classes existed before . I didn't want to change too much. We can probably have . a refactoring later